annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -515,19 +521,40 @@ private String resolveEntrezGeneId(boolean replace) {
         if (!replace || canonicalTranscript == null || canonicalTranscript.getGeneId() == null || canonicalTranscript.getGeneId().trim().isEmpty()) {
             return mRecord.getENTREZ_GENE_ID();
         }
+        // check hashmap first before hitting api for gene cross-ref data
+        if (ensemblAccessionEntrezIdMap.containsKey(canonicalTranscript.getGeneId())) {
+            return ensemblAccessionEntrezIdMap.get(canonicalTranscript.getGeneId());
+        }
+        
         // make the rest call to ensembl server for gene external refs
         RestTemplate restTemplate = new RestTemplate();
         HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = getRequestEntity();        
         ResponseEntity<GeneXref[]> responseEntity = restTemplate.exchange(geneXrefsServiceUrl + canonicalTranscript.getGeneId(), HttpMethod.GET, requestEntity, GeneXref[].class);
         
         String entrezGeneId = null;
+        List<GeneXref> geneXrefs = new ArrayList();
         for (GeneXref xref : responseEntity.getBody()) {
-            if (xref.getDbname().equals("EntrezGene")) {
-                entrezGeneId = xref.getPrimaryId();
-                break;
+            if (xref.getDbname().equals("EntrezGene") && xref.getDisplayId().equals(canonicalTranscript.getGeneSymbol())) {
+                geneXrefs.add(xref);
             }
         }
-        return entrezGeneId != null ? entrezGeneId : mRecord.getENTREZ_GENE_ID();
+        if (geneXrefs.size() > 1 ) {
+            for (GeneXref xref : geneXrefs) {
+                if (!xref.getDescription().toLowerCase().contains("pseudogene") && !xref.getDescription().toLowerCase().contains("uncharacterized")) {
+                    entrezGeneId = xref.getPrimaryId();
```

Comment:
Here it seems that if multiple entrez genes [ignoring pseudo genes and uncharacterized] map to the ensembl id, we are going to be taking the first one encountered in the body, and not logging the fact that others were ignored. Maybe this cannot happen?